ext/nnstreamer/tensor_converter/tensor_converter_flatbuf.cc

```diff
@@ -0,0 +1,169 @@
+ * Copyright (C) 2020 Gichan Jang <gichan2.jang@samsung.com>
+ */
+ /**
+
+ * @file        tensor_converter_flatbuf.cc
+ * @date        14 May 2020
+ * @brief       NNStreamer tensor-converter subplugin, "flatbuffer",
+ *              which converts flatbuufers byte stream to tensors.
+ * @see         https://github.com/nnsuite/nnstreamer
+ * @author      Gichan Jang <gichan2.jang@samsung.com>
+ * @bug         No known bugs except for NYI items
+ *
+ */
+
+ /**
+  * Install flatbuffers
+  * We assume that you use Ubuntu linux distribution.
+  * You may simply download binary packages from PPA
+  *
+  * $ sudo apt-add-repository ppa:nnstreamer
+  * $ sudo apt update
+  * $ sudo apt install libflatbuffers libflatbuffers-dev flatbuffers-compiler
+  */
+
+#include <iostream>
+#include <fstream>
+#include <typeinfo>
+#include <glib.h>
+#include <gst/gstinfo.h>
+#include <nnstreamer_plugin_api.h>
+#include "nnstreamer_plugin_api_converter.h"
+#include <nnstreamer_log.h>
+#include <nnstreamer_generated.h>       // Generated by `flatc`.
+
+namespace nnstreamer {
+namespace flatbuf {
+
+void init_fbc (void) __attribute__ ((constructor));
+void fini_fbc (void) __attribute__ ((destructor));
+
+/** @brief tensor converter plugin's NNStreamerExternalConverter callback */
+static GstCaps *
+fbc_query_caps (const GstTensorsConfig * config)
+{
+  return gst_caps_from_string (GST_FLATBUF_TENSOR_CAP_DEFAULT);
+}
+
+/** @brief tensor converter plugin's NNStreamerExternalConverter callback */
+static gboolean
+fbc_get_out_config (const GstCaps * in_cap, GstTensorsConfig * config)
+{
+  GstStructure *structure;
+  g_return_val_if_fail (config != NULL, FALSE);
+  gst_tensors_config_init (config);
+  g_return_val_if_fail (in_cap != NULL, FALSE);
+
+  structure = gst_caps_get_structure (in_cap, 0);
+  g_return_val_if_fail (structure != NULL, FALSE);
+
+  /* All tensor info should be updated later in chain function. */
+  config->info.info[0].type = _NNS_UINT8;
+  config->info.num_tensors = 1;
+  if (gst_tensor_parse_dimension ("1:1:1:1",
+          config->info.info[0].dimension) == 0) {
+    ml_loge ("Failed to set initial dimension for subplugin");
+    return FALSE;
+  }
+
+  if (gst_structure_has_field (structure, "framerate")) {
+    gst_structure_get_fraction (structure, "framerate", &config->rate_n,
+        &config->rate_d);
+  } else {
+    /* cannot get the framerate */
+    config->rate_n = 0;
+    config->rate_d = 1;
+  }
+  return TRUE;
+}
+
+/** @brief tensor converter plugin's NNStreamerExternalConverter callback
+ *  @todo : Consider multi frames, return Bufferlist and 
+ *          remove frame size and the number of frames
+ */
+static GstBuffer *
+fbc_convert (GstBuffer * in_buf, gsize * frame_size, guint * frames_in,
+    GstTensorsConfig * config)
+{
+  const Tensors *tensors;
+  const flatbuffers::Vector < flatbuffers::Offset < Tensor >> *tensor;
+  const flatbuffers::Vector < unsigned char >*tensor_data;
```

Comment:
I don't think you need to add spaces for template declarations (e.g., ```<unsigned char>``` is fine). Don't apply gst-indent for C++ (.cc) files.

Reply:
Updated!
Additional question : 
When I applied clang-format, there is no space between the function name and arguments.
 e.g : `in_mem = gst_buffer_peek_memory(in_buf, 0);` 
Is it intended result of clang formatter?